index.d.ts

```diff
@@ -48,62 +48,78 @@ export interface State {
 
 export type Href = string | Location;
 
-export const LOCATION_CHANGED: string;
-export const PUSH: string;
-export const REPLACE: string;
-export const GO: string;
-export const GO_BACK: string;
-export const GO_FORWARD: string;
-export const POP: string;
-export const REPLACE_ROUTES: string;
-export const DID_REPLACE_ROUTES: string;
-
-export function push(href: Href, options: LocationOptions): {
-  type: string;
+export const LOCATION_CHANGED = 'ROUTER_LOCATION_CHANGED';
```

Comment:
In hindsight, I wonder whether typescript can still narrow the types if we leave the actual value out of here (just like it was before). This might allow for greater implementation flexibility, because you'll then be free to change the action types as you wish.. I need to figure this out.